shared/src/cell_set.rs

```diff
@@ -65,6 +65,11 @@ pub struct CellSet {
     pub(crate) inner: FnvHashMap<H256, TransactionMeta>,
 }
 
+pub(crate) enum CellSetOpr {
+    Delete,
+    Update(TransactionMeta),
```

Comment:
is it possible to use lifetime here and voiding clone in `mark_dead`?

Reply:
I can return `E<&T>`, but I can't put some `&T` from `E<&T>` into `Vec<&T>`.

Or maybe I can use `&T` immediately, avoid `Vec<&T>`.

But the code will be **very hard to understand**: I have to mix `insert` and `remove` both in memory and database, in one closure.

I tried before I pulled this PR, but I failed, I asked @driftluo for help, and he think that is impossible, too.

Reply:
Order is important, all `remove` have to be after all `insert`, so, I have to collect all `insert` and `remove` before do it.

So, I can't avoid `Vec<&T>`.

Reply:
Get, thanks.